sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -215,3 +231,21 @@ def _prepare_new_so_line(self, line_form, sequence):
         line_form.product_id = self.product_id
         line_form.sequence = sequence
         line_form.product_uom_qty = self.units_included
+
+    def _get_last_sale_price_product(self):
+        """
+        Get last price from last order.
+        Use sudo to read sale order from other users like as other commercials.
+        """
+        self.ensure_one()
+        so = self.env["sale.order"].sudo().search([
+            ("partner_id", "=", fields.first(self).partner_id.id),
```

Comment:
first() method is unneeded If you use ensure_one
```suggestion
            ("partner_id", "=", self.partner_id.id),
```